index.bs

```diff
@@ -3922,6 +3936,30 @@ Note: The {{UserVerificationRequirement}} enumeration is deliberately not refere
 </div>
 
 
+### User-agent Hints Enumeration (enum <dfn enum>PublicKeyCredentialHints</dfn>) ### {#enum-hints}
+
+<xmp class="idl">
+    enum PublicKeyCredentialHints {
+        "security-key",
+        "non-security-key",
+    };
+</xmp>
+
+Note: The {{PublicKeyCredentialHints}} enumeration is deliberately not referenced, see [[#sct-domstring-backwards-compatibility]].
+
+<div dfn-type="enum-value" dfn-for="PublicKeyCredentialHints">
+    [=[WRPS]=] may use this enumeration to communicate hints to the user-agent about how a request may be best completed. These hints are not requirements, and do not bind the user-agent, but may guide it in providing the best experience by using contextual information that the [=[RP]=] has about the request. Hints are provided in order of decreasing preference so, if two hints are contradictory, the first one controls. Hints may also overlap: if a more-specific hint is defined a [=[RP]=] may still wish to send less specific ones for user-agents that may not recognise the more specific one. In this case the most specific hint should be sent before the less-specific ones.
+
+    Hints may contradict information in contained in credential {{PublicKeyCredentialDescriptor/transports}} value controls. (Although when using [=discoverable credentials=], {{PublicKeyCredentialDescriptor/transports}} values are not provided leaving hints as the only avenue for expressing some aspects of the request.)
```

Comment:
```suggestion
    Hints MAY contradict information contained in credential {{PublicKeyCredentialDescriptor/transports}} value controls. (Although when using [=discoverable credentials=], {{PublicKeyCredentialDescriptor/transports}} values are not provided leaving hints as the only avenue for expressing some aspects of the request.)
```
Dunno if the "MAY" is the fancy kind or not.

Reply:
(Applied outside of GitHub because GitHub can't merge multiple suggestions in the same line.)